sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -208,11 +210,30 @@ private[sql] object ParquetFilters {
   }
 
   /**
+   *  Return referenced columns in [[sources.Filter]].
+   */
+  def referencedColumns(schema: StructType, predicate: sources.Filter): Array[String] = {
```

Comment:
Oh, yes it looks so. I think I might also have to change `createFilter()` in that way because I just followed up in the way of `createFilter()` for this function because both `createFilter()` and `referencedColumns()` are called in the same places.